app/models/address.rb

```diff
@@ -25,4 +33,22 @@ def to_a
       postcode
     ].map(&:presence).compact
   end
+
+  private
+
+  def set_point_from_postcode
+    return if postcode.blank? || point.present?
+    geocode = geocode_postcode
+    if geocode && geocode.data['address_components'].any? { |a| a['types'] == ['postal_code'] }
+      self.point = {
+        type: 'Point',
+        coordinates: [geocode.longitude, geocode.latitude]
+      }
+    end
+  end
+
+  def geocode_postcode
+    results = Geocoder.search(postcode)
+    results.count == 1 ? results.first : nil
```

Comment:
This is designed to block the case when more than one result is returned. Which will result is the address change failing. In this case, the address will need to be manually set by us for the time being. Maybe wrapping the Geocoder with a decorator class to give it a nicer interface would make this clearer?